sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -1299,8 +1299,8 @@ class SessionCatalog(
 
       // Check input argument size
       if (e.inputTypes.size != input.size) {
-        throw new AnalysisException(s"Invalid number of arguments for function $name. " +
-          s"Expected: ${e.inputTypes.size}; Found: ${input.size}")
+        throw new InvalidFunctionArgumentException(s"Invalid number of arguments for " +
```

Comment:
Instead of using a special exception here, maybe it's better to create a `InvalidUDFClassException` for the else branch, as it's not really a plan analysis error.

Reply:
> Instead of using a special exception here, maybe it's better to create a `InvalidUDFClassException` for the else branch, as it's not really a plan analysis error.

Raise a new pr or just fix this in this pr

Reply:
> Instead of using a special exception here, maybe it's better to create a `InvalidUDFClassException` for the else branch, as it's not really a plan analysis error.

So you mean add a exception of 
```
    } else {
      throw new AnalysisException(s"No handler for UDAF '${clazz.getCanonicalName}'. " +
        s"Use sparkSession.udf.register(...) instead.")
    }
```

Reply:
yea

Reply:
and `InvalidUDFClassException` doesn't need to be a public class.

Reply:
> and `InvalidUDFClassException` doesn't need to be a public class.

emmmm， see latest change,  maybe I miss understand what you really want to do.